src/entities/BaseEntity.java

```diff
@@ -0,0 +1,67 @@
+package entities;
+
+import java.util.Date;
+
+public abstract class BaseEntity<T> {
+    private Class<T> entity;
```

Comment:
What is this for?

Reply:
It's an attribute that contains the object that will be traced, but since the other objects will extend from BaseEntity i see now that it probably wont be necessary

Reply:
Are tou working on trace structure? If that is the case there couple of things missing.

I think you don't want to mix tracing architecture with entities structure.

Reply:
What i mean by tracing is that this entity will contain when an object is created and updated, and by who

Reply:
Why would you create and abstract class to hold the entity that is being created/updated?

Why not to just extended from the entity class itself?

Reply:
I'll remove the Class<T> entity attribute in BaseEntity since it's not necessary, but the other classes will extend from Base Entity, the class User, for example, extends from BaseEntity

Reply:
Then I think also adding the id is good as well, since every single entity has an id

Reply:
Ok no problem, i'll do that